App/Sagas/TextileSagas.ts

```diff
@@ -564,6 +565,120 @@ export function * refreshThreads () {
+    // This shouldn't be possible currently
+  } else {
+    try {
+      const image: TextileTypes.SharedImage = {
+        origURL: pickerResponse.origURL || pickerResponse.uri,
+        uri: pickerResponse.uri,
+        height: pickerResponse.height,
+        width: pickerResponse.width,
+        isVertical: pickerResponse.isVertical
+      }
+      yield put(CameraRollActions.imagePickerSuccess(threadId, image))
+      yield call(PhotosNavigationService.navigate, 'SharePhoto', {threadId, image})
+    } catch (error) {
+      yield put(UIActions.newImagePickerError(error, 'There was an issue with your photo selection. Please try again.'))
+    }
+  }
+​}
+
+export function * localPinRequest(action: ActionType<typeof CameraRollActions.addComment>) {
+  const {threadId, image} = action.payload
+  const photoPath = image.uri.replace('file://', '')
+  try {
+    // add the result to our local node
+    const addResult: TextileTypes.AddResult = yield call(TextileNode.addPhoto, photoPath)
+    if (!addResult.archive) {
+      throw new Error('No archive returned')
+    }
+
+    // Get the ID of our Default Thread
+    let defaultThread: TextileTypes.Thread = yield call(getDefaultThread)
+    // Add the image to our defaul thread // i.e the wallet
+    yield call(TextileNode.addPhotoToThread, addResult.id, addResult.key, defaultThread.id)
+
+    // Share the photo to the target Thread
+    yield call(TextileNode.sharePhotoToThread, addResult.id, threadId, image.caption)
+    // Notify the UI to update via updating the hashes
+    yield put(TextileNodeActions.getPhotoHashesRequest(threadId))
+
+    // Store the addResult with the image
+    yield put(CameraRollActions.localPinSuccess(threadId, image, addResult))
+  } catch (error) {
+    try {
+      yield put(CameraRollActions.imagePinError(threadId, image))
+      const exists: boolean = yield call(RNFS.exists, photoPath)
+      if (exists) {
+        yield call(RNFS.unlink, photoPath)
+      }
+    } finally {
+      yield put(UIActions.newImagePickerError(error, 'There was an issue with your local IPFS node. Please try again.'))
+    }
+  }
+}
+
+export function * remotePinRequest(action: ActionType<typeof CameraRollActions.localPinSuccess>) {
+  const {threadId, image, addResult} = action.payload
+  // There is a possibility that an image could successfully load into a users thread locally by
+  // here, but not succeed with 'UploadingImagesActions.addImage' so never upload
+  try {
+    if (!addResult.archive) {
+      throw new Error('no archive returned')
+    }
+    // Add the image upload job to the tracked jobs
+    yield put(UploadingImagesActions.addImage(addResult.archive.path, addResult.id, 3))
+
+    // Begin the image upload to the Cafe pinner
+    yield uploadFile(
+      addResult.id,
+      addResult.archive.path
+    )
+
+    // Remove the image from the CameraRoll actions
+    yield put(CameraRollActions.remotePinSuccess(threadId, image))
```

Comment:
At this point, the image hasn't finished uploading. If `CameraRollActions.remotePinSuccess` is intended to fire once the image is done uploading, it needs to happen once we're notified the upload is complete, similar to how we are doing it in `TextileManager`, firing `UploadingImagesActions.imageUploadComplete`.

Reply:
Right, the intention here was more, remotePinInitSuccess. As in, the problem is no longer the camerarolls, now it's over to the uploader